code/modules/mob/living/living.dm

```diff
@@ -1495,3 +1495,36 @@
 	if(!apply_change)
 		return BODYTEMP_NORMAL
 	return BODYTEMP_NORMAL + get_body_temp_normal_change()
+
+/mob/living/proc/can_look_up() //Checks if the user is incapacitated.
+	return !((next_move > world.time) || incapacitated(ignore_restraints = TRUE))
+
+/**
+ * look_up Changes the perspective of the mob to any openspace turf above the mob
+ *
+ * This also checks if an openspace turf is above the mob before looking up or resets the perspective if already looking up
+ *
+ */
+/mob/living/proc/look_up()
+
+	if(client.perspective != MOB_PERSPECTIVE) //We are already looking up.
+		stop_look_up()
+		return
+	if(!can_look_up())
+		return
+	var/turf/ceiling= get_step_multiz(src, UP)
+	if(!ceiling) //We are at the highest z-level.
+		to_chat(src, "<span class='warning'>You can't see through the ceiling above you.</span>")
+		return
+	else if(!isopenspace(ceiling)) //There is no openspace turf above us.
+		to_chat(src, "<span class='warning'>You can't see through the floor above you.</span>")
+		return
+
+	changeNext_move(CLICK_CD_LOOK_UP)
+	SEND_SIGNAL(src, COMSIG_LIVING_LOOK_UP, src)
```

Comment:
Why is this signal a thing? Also the argument is redundant.

Reply:
Apologies on the redundant signal argument.  I figured emitting a signal wouldn't be a bad idea in case someone else wanted tie some other functionality to a look up.  But it can easily be removed.